cloudimg/aws.py

```diff
@@ -413,14 +415,36 @@ def upload_to_container(self, image_path, container_name, object_name,
         if image_path.lower().startswith('http'):
             # Stream the upload from a remote URL
             log.info('Opening stream to: %s', image_path)
-            resp = requests.get(image_path, stream=True)
+            resp = requests.get(image_path, stream=True, timeout=30)
             resp.raise_for_status()
             stream = resp.iter_content(chunk_size)
             callback = UploadProgress(container_name, object_name)
-            self.s3.meta.client.upload_fileobj(stream,
-                                               container_name,
-                                               object_name,
-                                               Callback=callback)
+            if object_name.endswith(".xz"):
+                object_name = object_name.rstrip(".xz")
+                log.info(
+                    "Processing a remote LZMA compressed file: %s",
+                    object_name)
+                stream_lzma = lzma.LZMAFile(stream)
```

Comment:
That line will very likely not work.
`resp.iter_content` returns generator/iterable and it cannot be used as input for lzma.LZMAFile

You would need to implement something like it was mentioned in the other MR
* https://github.com/release-engineering/cloudimg/pull/18#discussion_r1130712138

I would personal raise NotImplementedError  if we detect compressed (`xz`) format with http for image path

I do not know how this code path was used before. So we probably do not want to drop it as in azure MR :-)

Reply:
Added